### PR TITLE
Add reading of secrets.config file

### PR DESCRIPTION
**Pull request by alexpvpmindustry**

### CODE REVIEW

**Comment by alexpvpmindustry** [comment]
solves #87

File: AI-InvestiBot/implementation.py [hunk 1]
```diff
@@ -25,15 +25,38 @@
...
+    raise
+
+if YOUR_API_KEY_ID is None or YOUR_API_KEY_ID=="":
+    raise ValueError("Set your API key ID from alpaca")
+if YOUR_SECRET_KEY is None or YOUR_SECRET_KEY=="":
+    raise ValueError("Set your secret key from alpaca")
+if BUCKET_NAME is None or BUCKET_NAME=="":
+    raise ValueError("Set your BUCKET_NAME from AWS")
+if OBJECT_KEY is None or OBJECT_KEY=="":
+    raise ValueError("Set your OBJECT_KEY from AWS")
```
Comment (gran4) [inline AI-InvestiBot/implementation.py:58]: Not all people will want to do both aws.

Comment (alexpvpmindustry) [reply AI-InvestiBot/implementation.py:57]: True... But currently the code uses boto3 and isn't boto3 from AWS?
Personally I hate AWS, but this project seems to heavily rely on it...

Takes forever to configure the AWS console correctly and I'm still troubleshooting it.

For json storage I prefer mongodb, atlas has a free tier up to 500mb, it's basically a "dictionary on the cloud"

I have no alternatives for AWS lambda tho, other than my own servers (which has very little compute)

Comment (gran4) [reply AI-InvestiBot/implementation.py]: Not the project does not rely on aws, was is just an experimental way of running the bot without a laptop open. Sure we can switch it to something else.

P.S: If you do not want the lambda version, run the loop implementation.
There are 2 implementations: the aws, and the loop(just a while True loop).

Comment (gran4) [reply AI-InvestiBot/implementation.py]: mongodb might be cheaper as well.

**Comment by alexpvpmindustry** [comment]
how do i do the while true loop? @gran4

File: AI-InvestiBot/implementation.py [hunk 2]
```diff
@@ -165,16 +177,19 @@ def update_models(models, total_info_keys):
...
+            manager.sell()
         if profit < RISK_REWARD_RATIO:
             break
-        RESOURCE_MANAGER.buy(model[0].stock_symbol)
+        manager.buy(model[0].stock_symbol)
         i += 1
 
 
-def run_loop(models, total_info_keys) -> None:
+def run_loop(models, total_info_keys, manager: Optional[ResourceManager]=None) -> None:
```
Comment (alexpvpmindustry) [inline AI-InvestiBot/implementation.py:188]: i see the while true loop here, but manager needs to be specified,
if not it will be using AWS.

**Comment by alexpvpmindustry** [comment]
possibly, the simplest way is to add a `ResourceManagerLocal` that is only local

and rename the current one to `ResourceManagerAWS`

**Comment by alexpvpmindustry** [comment]
so have `ResourceManager` be a interface to create `ResourceManagerLocal` and `ResourceManagerAWS`

File: AI-InvestiBot/implementation.py (hunk 1 above)
Comment (gran4) [reply AI-InvestiBot/implementation.py]: Also, I can easily add more implementations

File: AI-InvestiBot/implementation.py (hunk 2 above)
Comment (gran4) [reply AI-InvestiBot/implementation.py:188]: ?
The while true loop is inside of the function. If the manager is None, the project will set the manager.

**Comment by alexpvpmindustry** [comment]
nvm... i got confused... `ResourceManager` is basically alpaca rest API.

**Comment by gran4** [comment]
> possibly, the simplest way is to add a `ResourceManagerLocal` that is only local
> 
> and rename the current one to `ResourceManagerAWS`
> 
> ie, `ResourceManager` interface, to create `ResourceManagerAWS` and `ResourceManagerLocal`

Not needed. I made it so that this PR makes it so that you put the resource manager as a parameter. It is just better in like every way..

**Pull request merged**